src/apps/vpn/controller.cpp

```diff
@@ -219,7 +220,38 @@ bool Controller::activate(const ServerData& serverData,
       m_portalDetected = false;
       return true;
     }
-    setState(StateConnecting);
+
+    // Before attempting to enable VPN connection we should check that the
+    // subscription is active.
+    setState(StateCheckSubscription);
+
+    // Set up a network request to check the subscription status.
+    // "task" is an empty task function which is being used to
+    // replicate the behavior of a TaskAccount.
+    TaskFunction* task = new TaskFunction([]() {});
+    NetworkRequest* request = new NetworkRequest(task, 200);
+    request->auth(MozillaVPN::authorizationHeader());
+    request->get(AppConstants::apiUrl(AppConstants::Account));
+
+    connect(request, &NetworkRequest::requestFailed, this,
+            [this](QNetworkReply::NetworkError error, const QByteArray&) {
+              logger.error() << "Account request failed" << error;
+              REPORTNETWORKERROR(error, ErrorHandler::DoNotPropagateError,
+                                 "PreActivationSubscriptionCheck");
+              if (MozillaVPN::instance()->state() == MozillaVPN::StateMain) {
```

Comment:
Write a comment saying that, we need to see if the error propagation has changed the Mozilla VPN state. We continue only if the user is still authenticated and subscribed.